app/models/miq_worker.rb

```diff
@@ -40,6 +40,10 @@ class MiqWorker < ApplicationRecord
 
   PROCESS_TITLE_PREFIX = "MIQ:".freeze
 
+  after_initialize do |worker|
+    worker.system_uid = worker.unit_name if worker.systemd_worker?
+  end
```

Comment:
Minor but in the past, we've tried to avoid using the block form for callbacks instead preferring a private method, making this something like:

```suggestion
  after_initialize :set_system_uid
  
  private def set_system_uid
    self.system_uid = unit_name if systemd_worker?
  end
```